### ISSUE TITLE
It’s not always accurate.

### ISSUE THREAD

**Issue by Dark-Aura** [projectdiscovery/urlfinder#19]
![Image](https://github.com/user-attachments/assets/03b2a263-361f-4d8d-a85e-f232d2ce4a22)

I’m running the same command, but I’m getting different output results. How can I overcome this issue? The output isn’t accurate each time. For example, before adding the `URLscan` API key, the output was **134707**.

![Image](https://github.com/user-attachments/assets/e24f7d12-e473-4f82-817f-507e3fafc17f)

After adding the `URLscan` API key to “provider-config.yaml,” I’m receiving even fewer URLs. Why is this happening?

![Image](https://github.com/user-attachments/assets/4aab6075-d1d9-453b-aa6d-1ebd1605a93f)

**Comment by zy9ard3** [comment]
@dogancanbakir 

I confirm this is indeed an issue with `urlscan` and not with statistics

```console
┌──(root🍀zy9ard3)-[~]
└─# wc -l withapikey withoutapikey 
   95766 withapikey
  171594 withoutapikey
  267360 total
```

**Comment by dogancanbakir** [comment]
@zy9ard3 Are you suggesting that there is no issue with `urlfinder` ?

**Comment by zy9ard3** [comment]
@dogancanbakir 

> [@zy9ard3](https://github.com/zy9ard3) Are you suggesting that there is no issue with `urlfinder` ?

No, what I meant is that the issue arises from the `urlscan` implementation and no issue with the statistics

I verified both the outputs ( with and without the `urlscan` API key ) and found that some URLs are being missed when using the `urlscan` API key

**Comment by dogancanbakir** [comment]
Thanks, I'll look into it.

**Comment by gwatai** [comment]
Hello folks, I hope you are doing great, Any update on the issue? Thank you
